### ISSUE TITLE
Application Error after update to RocketChat 6.3.1

### ISSUE THREAD

**Issue by larrymarquez** [RocketChat/Rocket.Chat#30097]
### Description:
UI crashes when trying to edit a User
Also I get: 
Error
Oops! Something went wrong. Please reload the page or contact an administrator.

### Steps to reproduce:

1.  A chat Room 
2.  Click to open chat
3. 

### Actual behavior: I get: 
Error
Oops! Something went wrong. Please reload the page or contact an administrator.
When i go to Edit a user i get:
Application Error
The application GUI just crashed.
![image](https://github.com/RocketChat/Rocket.Chat/assets/89091898/918f52ae-638d-41c5-80e4-325833aebe5c)
![image](https://github.com/RocketChat/Rocket.Chat/assets/89091898/643bbe80-3452-4233-a4fd-4e9b3b3ad6a8)

### Server Setup Information:

- Version of Rocket.Chat Server: 6.3.1
- Operating System: Ubuntu 22.04
- Deployment Method: snap
- Number of Running Instances: 1
- DB Replicaset Oplog: 
- NodeJS Version:  v14.21.2
- MongoDB Version: 5.0.15 / wiredTiger (oplog Habilitada)

### Client Setup Information

- Desktop App or Browser Version: Chrome, desktop app
- Operating System: Windows

**Comment by akong77** [comment]
I has got the same error message.

**Comment by monchulanderas** [comment]
Same error

**Comment by lovekyleman** [comment]
Same error

**Comment by iomismo** [comment]
Same here. 
Console:
```
peError: Cannot read properties of undefined (reading 'hours')
    at lastDay (eval at <anonymous> (TranslationProvider.tsx:1:1), <anonymous>:51:46)
    at A.R [as calendar] (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:225:719703)
    at 241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2860279
    at b (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2163978)
    at Ii (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:59708)
    at $c (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:111738)
    at nu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98852)
    at tu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98780)
    at eu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98643)
    at Hc (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:95497)
```

**Comment by monchulanderas** [comment]
Firefox console:

TypeError: can't access property "hours", this is undefined
    lastDay https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true line 327 > eval line 1 > Function:51
    R https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:225
    r https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522
    w /client/components/message/content/ThreadMetrics.tsx:1
    Ii https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    $c https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    nu https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    tu https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    eu https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    Hc https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    xa https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    unstable_runWithPriority https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:166
    _a https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    xa https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    va https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    Fc https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    rs https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    f https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:111
    c https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522
    c https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522
241ed9ab96b707efb232a05e041797fec786ca1e.js:130:82908
    zs https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    payload https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    La https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    gs https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    $c https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    nu https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    tu https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    eu https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    Hc https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    xa https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    unstable_runWithPriority https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:166
    _a https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    xa https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    va https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    Fc https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    rs https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130
    f https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:111
    c https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522
    c https://dominio.com/241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522

**Comment by monchulanderas** [comment]
> 


> > Same here. Console:
> > ```
> > peError: Cannot read properties of undefined (reading 'hours')
> >     at lastDay (eval at <anonymous> (TranslationProvider.tsx:1:1), <anonymous>:51:46)
> >     at A.R [as calendar] (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:225:719703)
> >     at 241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2860279
> >     at b (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2163978)
> >     at Ii (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:59708)
> >     at $c (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:111738)
> >     at nu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98852)
> >     at tu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98780)
> >     at eu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98643)
> >     at Hc (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:95497)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Changing language to english works fine.
> 
> Hi, where did you change language ? from terminal or UI?


If you change the language of the browser and set 'English', then it works fine...

**Comment by akong77** [comment]
I use chinese language..Could fix it?

**Comment by larrymarquez** [comment]
> 找到解决方案了 把/opt/Rocket.Chat/programs/server/assets/app/i18n/zh-HK.json 备份 然后把zh.i18n.json 或者zh-CN.json 重命名为zh-HK.json 注意文件权限要rocketchat:rocketchat 到设置里语言设置成中文(香港) 个人偏好改成默认 就 不会再崩溃了 不知道为什么文件内容一样 但是用zh 或者zh-CN 就会崩溃 最后注意操作完最好清理掉浏览器缓存 @akong77 @iomismo @monchulanderas @larrymarquez @lovekyleman

Hi liyue,

  I have a snap installation of the rocketchat server. 
My server installation path is **/var/snap/rocketchat-server**

Does this work the same in snap installation mode?

Thank you!

**Comment by liyue1010110** [comment]
use find to get the language file *.json ; cp to the location; in normal backup en.i18n.json and rename (you language).json to  en.i18n.json;  go web page settings/General find to language change to en  ; clean the web cache and reload the page

**Comment by rccardoso** [comment]
> Same here. Console:
> 
> ```
> peError: Cannot read properties of undefined (reading 'hours')
>     at lastDay (eval at <anonymous> (TranslationProvider.tsx:1:1), <anonymous>:51:46)
>     at A.R [as calendar] (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:225:719703)
>     at 241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2860279
>     at b (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2163978)
>     at Ii (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:59708)
>     at $c (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:111738)
>     at nu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98852)
>     at tu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98780)
>     at eu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98643)
>     at Hc (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:95497)
> ```
> 
> Changing language to english works fine.

Yes, here also.

**Comment by macanatta** [comment]
I have the same problem with Spanish

**Comment by iomismo** [comment]
> > Same here. Console:
> > ```
> > peError: Cannot read properties of undefined (reading 'hours')
> >     at lastDay (eval at <anonymous> (TranslationProvider.tsx:1:1), <anonymous>:51:46)
> >     at A.R [as calendar] (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:225:719703)
> >     at 241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2860279
> >     at b (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:522:2163978)
> >     at Ii (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:59708)
> >     at $c (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:111738)
> >     at nu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98852)
> >     at tu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98780)
> >     at eu (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:98643)
> >     at Hc (241ed9ab96b707efb232a05e041797fec786ca1e.js?meteor_js_resource=true:130:95497)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Changing language to english works fine.
> 
> Hi, where did you change language ? from terminal or UI?

UI > Settings > General

I had problems with some rooms (Oops!) and Application Error GUI on user editing, but Settings worked so I had no problem changing language.

**Comment by monchulanderas** [comment]
> use find to get the language file *.json ; cd to the location; in normal backup en.i18n.json and rename (you language).json to en.i18n.json; go web page settings/General find to language change to en ; clean the web cache and reload the page @larrymarquez


it worked! Thanks...

**Comment by akong77** [comment]
Change to EN is should temp solved.Should this issue be fixed in the future?

**Comment by lovekyleman** [comment]
> should
Believe me. Will take a long time to fix it

**Comment by akong77** [comment]
OK,I will wait it.Thanks a lot.

**Comment by chaoslawful** [comment]
See #30082

**Comment by gusarg81** [comment]
By the way, it does happen with Spanish too as well.

**Comment by florisvangeel** [comment]
confirmed there is some serious regression or broken since 6.3.x
Upto 6.2.12 all was working fine in building the package as run dev.
```
@rocket.chat/cron:dev: ERROR: command finished with error: command (/home/%user%/rocketchat/packages/cron) yarn run dev exited (129)
command (/home/%user%/rocketchat/packages/cron) yarn run dev exited (129)

 Tasks:    0 successful, 29 total
Cached:    0 cached, 29 total
  Time:    53.128s 
Failed:    @rocket.chat/cron#dev

 ERROR  run failed: command  exited (129)
 ```
node -v
v14.21.3
ubuntu 20.04 with all dependencies working upto 6.2.12

**Comment by larrymarquez** [comment]
Changing the language to English, worked. From User profile preferences,  I
hope this issue can be fixed in the next release.

**Comment by florisvangeel** [comment]
This issue seems fixed in 6.3.2

**Comment by lovekyleman** [comment]
> This issue seems fixed in 6.3.2

No

**Comment by hugocostadev** [comment]
Seems to be fixed now in the latest release, could you check, please? Thanks

---
**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there
- Feature Request: Open an issue in [Feature Request](https://github.com/RocketChat/feature-requests) repository

**Comment by akong77** [comment]
> > This issue seems fixed in 6.3.2
> 
> No

Yes,I upgrade to 6.3.3.It's will show the same error line screenshot.
![圖片](https://github.com/RocketChat/Rocket.Chat/assets/6081927/365d3035-291d-4f0f-af12-7ef4ecb2ec68)

**Comment by akong77** [comment]
Then I change language to default.It's will to normal.Then I change language to trad. chinese.It's also no error.I will try it.

**Comment by macanatta** [comment]
According to [#30082](https://github.com/RocketChat/Rocket.Chat/issues/30082). The problem is related to quotes. Chats containing quoted messages do not work in languages different from english.

**Comment by yash-rajpal** [comment]
Thanks for reporting the bug, this one is fixed by #30167 and the fix will be available soon as a patch release for version 6.3.
